packages/ovm/src/app/utils.ts

```diff
@@ -61,34 +62,66 @@ export interface OvmTransactionMetadata {
 }
 
 /**
- * Convert internal logs into OVM logs. Or in other words, take the logs which
+ * Convert internal transaction logs into OVM logs. Or in other words, take the logs which
  * are emitted by a normal Ganache or Geth node (this will include logs from the ExecutionManager),
  * parse them, and then convert them into logs which look like they would if you were running this tx
  * using an OVM backend.
  *
+ * NOTE: The input logs MUST NOT be stripped of any Execution Manager events, or this function will break.
  *
- * @param logs an array of internal logs which we will parse and then convert.
+ * @param logs an array of internal transaction logs which we will parse and then convert.
  * @return the converted logs
  */
-export const convertInternalLogsToOvmLogs = (logs: Log[]): Log[] => {
+export const convertInternalLogsToOvmLogs = (
+  logs: Log[],
+  executionManagerAddress: string
+): Log[] => {
   let activeContract = logs[0] ? logs[0].address : ZERO_ADDRESS
-  const loggerLogs = [`Parsing logs from contract ${activeContract}: `]
+  const loggerLogs = [`Parsing internal logs ${JSON.stringify(logs)}: `]
   const ovmLogs = []
+  let cumulativeTxEMLogIndices = 0
+  let prevEMLogIndex = 0
   logs.forEach((log) => {
-    const executionManagerLog = executionManagerInterface.parseLog(log)
-    if (executionManagerLog) {
-      if (executionManagerLog.name === 'ActiveContract') {
-        activeContract = executionManagerLog.values['_activeContract']
+    if (log.address.toUpperCase() === executionManagerAddress.toUpperCase()) {
+      const EMLogIndex = log.logIndex
+      if (EMLogIndex <= prevEMLogIndex) {
+        loggerLogs.push(
+          `Detected raw EM log ${log} with lower logIndex than previously processed, must be from a new transaction.  Resetting cumulative EM log indices for tx.`
+        )
+        cumulativeTxEMLogIndices = 0
+      }
+      cumulativeTxEMLogIndices++
+      prevEMLogIndex = EMLogIndex
+      const executionManagerLog = executionManagerInterface.parseLog(log)
+      if (!executionManagerLog) {
+        loggerLogs.push(
+          `execution manager log ${log} was unrecognized by the interface parser--Definitely not an activeContract event, ignoring...`
```

Comment:
I think ${log} will just print `[object Object]`, but `JSON.stringify(...)` will result in circular error. Is there another property that would be useful to log here? `log.name`?